sympy/parsing/c/c_parser.py

```diff
@@ -150,7 +153,7 @@ def parse_str(self, source, flags):
                 A list of sympy AST nodes
 
             """
-            file = tempfile.NamedTemporaryFile(mode = 'w+', suffix = '.h')
+            file = tempfile.NamedTemporaryFile(mode = 'w+', suffix = '.cpp')
```

Comment:
How about creating two functions `parse_c_str` and `parse_cpp_str`, one with a `.c` extenstion and the other with a `.cpp` extension. 
But that can be done better in another PR, as we can also change the API a little accordingly.

Reply:
Sure that will be better. Will discuss it in a new PR...